x-pack/plugins/cases/server/client/attachments/delete.ts

```diff
@@ -0,0 +1,154 @@
+  subCaseID?: string;
+}
+
+/**
+ * Delete all comments for a case or sub case.
+ */
+export async function deleteAll(
+  { caseID, subCaseID }: DeleteAllArgs,
+  clientArgs: CasesClientArgs
+): Promise<void> {
+  const {
+    user,
+    savedObjectsClient: soClient,
+    caseService,
+    attachmentService,
+    userActionService,
+    logger,
+  } = clientArgs;
+
+  try {
+    checkEnabledCaseConnectorOrThrow(subCaseID);
+
+    const id = subCaseID ?? caseID;
+    const comments = await caseService.getCommentsByAssociation({
+      soClient,
+      id,
+      associationType: subCaseID ? AssociationType.subCase : AssociationType.case,
+    });
+
+    await Promise.all(
+      comments.saved_objects.map((comment) =>
+        attachmentService.delete({
+          soClient,
+          attachmentId: comment.id,
+        })
+      )
+    );
+
+    const deleteDate = new Date().toISOString();
+
+    await userActionService.bulkCreate({
+      soClient,
+      actions: comments.saved_objects.map((comment) =>
+        buildCommentUserActionItem({
+          action: 'delete',
+          actionAt: deleteDate,
+          actionBy: user,
+          caseId: caseID,
+          subCaseId: subCaseID,
+          commentId: comment.id,
+          fields: ['comment'],
+        })
+      ),
+    });
+  } catch (error) {
+    throw createCaseError({
+      message: `Failed to delete all comments case id: ${caseID} sub case id: ${subCaseID}: ${error}`,
+      error,
+      logger,
+    });
+  }
+}
+
+export async function deleteComment(
+  { caseID, attachmentID, subCaseID }: DeleteArgs,
+  clientArgs: CasesClientArgs
+) {
+  const {
+    user,
+    savedObjectsClient: soClient,
+    attachmentService,
+    userActionService,
+    logger,
+  } = clientArgs;
+
+  try {
+    checkEnabledCaseConnectorOrThrow(subCaseID);
+
+    const deleteDate = new Date().toISOString();
+
+    const myComment = await attachmentService.get({
+      soClient,
+      attachmentId: attachmentID,
+    });
+
+    if (myComment == null) {
+      throw Boom.notFound(`This comment ${attachmentID} does not exist anymore.`);
+    }
+
+    const type = subCaseID ? SUB_CASE_SAVED_OBJECT : CASE_SAVED_OBJECT;
+    const id = subCaseID ?? caseID;
+
+    const caseRef = myComment.references.find((c) => c.type === type);
+    if (caseRef == null || (caseRef != null && caseRef.id !== id)) {
+      throw Boom.notFound(`This comment ${attachmentID} does not exist in ${id}).`);
+    }
+
+    await attachmentService.delete({
+      soClient,
+      attachmentId: attachmentID,
+    });
+
+    await userActionService.bulkCreate({
+      soClient,
+      actions: [
+        buildCommentUserActionItem({
+          action: 'delete',
+          actionAt: deleteDate,
+          actionBy: user,
+          caseId: id,
+          subCaseId: subCaseID,
+          commentId: attachmentID,
+          fields: ['comment'],
+        }),
+      ],
+    });
+  } catch (error) {
+    throw createCaseError({
```

Comment:
Question about `createCaseError`: are these messages shown to users, including via HTTP responses? If so, I'm wondering if they should be internationalized.

Reply:
@oatkiller That's a good question. Yes, some of them are being shown to the users. I think `i18n` is not supported for the backend. Do you know how to internationalize this messages?

Reply:
Most of these messages are just logged but some of them do get bubbled up to the user. @XavierM @legrego @kobelb should we internationalize server side error and audit log messages?

Reply:
> should we internationalize server side error and audit log messages?

I don't know of an official stance, but personally I'd vote to keep them in English. That makes for a consistent auditing, debugging, and triaging experience, and makes searching for solutions based on error messages much easier

Reply:
@jonathan-buttner I don't think we have an official stance, but +1 for keeping them in English.

Reply:
I think we should still i18n the messages that are show to the user.

Reply:
@legrego @kobelb @jonathan-buttner @cnasikas I opened a discuss issue about this: https://github.com/elastic/kibana/issues/97650